packages/substrate-bindings/src/codecs/AccountId.ts

```diff
@@ -49,13 +44,25 @@ function fromBase58ToBuffer(nBytes: number) {
       checksum[1] !== expectedChecksum[1]
     )
       throw new Error("Invalid checksum")
+
+    if (toHex(prefixBytes) != toHex(getSs58FormatPrefixBytes(ss58Format)))
```

Comment:
please lets extract the number from the `prefixBytes` and then compare that number directly against the `ss58Format`? 🙏